### PR TITLE
Hosted CE doc (SOFTWARE-3445)

### PR DESCRIPTION
**Pull request by edquist**
Based on @matyasselmeci's #476.

### CODE REVIEW

File: docs/compute-element/hosted-ce.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+-   Validate operation with a set of test jobs
+
+-   Configure central OSG services to schedule jobs
+
+Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
+in order to support application software repositories.
+This will allow a broader set of jobs to run on your cluster.
+
+
+Technical Setup
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
Providing the OSG Worker Node client (HTCondor batch systems only)
```

File: docs/compute-element/hosted-ce.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+---------------
+
+An OSG Hosted CE site needs to have access to the worker node client.
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: All OSG sites need to provide the worker node client

File: docs/compute-element/hosted-ce.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+On a Hosted CE with a non-HTCondor batch system, this is often done by unpacking the worker node client tarball into the
+users home directories, if this is on a shared filesystem.
+But for sites with an HTCondor batch system, often there is no shared filesystem set up.
+In that case, the site can either set up a shared filesystem, or install the OSG worker node client.
+This can be obtained from the osg-wn-client rpm, the osg-wn tarball, or used directly from OASIS.
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Even though it's the next section, turn OASIS into a link to it

File: docs/compute-element/hosted-ce.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+in order to support application software repositories.
+This will allow a broader set of jobs to run on your cluster.
+
+
+Technical Setup
+---------------
+
+An OSG Hosted CE site needs to have access to the worker node client.
+On a Hosted CE with a non-HTCondor batch system, this is often done by unpacking the worker node client tarball into the
+users home directories, if this is on a shared filesystem.
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: I think we want to hide some of the gory details from the user and say something along the lines that this is normally handled by OSG staff in the hosted CE case but it requires home directories shared across the cluster.

File: docs/compute-element/hosted-ce.md (hunk 3 above)
Comment (brianhlin) [reply docs/compute-element/hosted-ce.md]: We should also say something along the lines of: if it is easier for your site to install and maintain the WN client than to set up a shared file system, you have the following options:

File: docs/compute-element/hosted-ce.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+But for sites with an HTCondor batch system, often there is no shared filesystem set up.
+In that case, the site can either set up a shared filesystem, or
+[install the OSG worker node client](/worker-node/install-wn).
+This can be obtained from the [osg-wn-client rpm](/worker-node/install-wn),
+the [osg-wn tarball](/worker-node/install-wn-tarball),
+or installed directly from [OASIS](/worker-node/install-wn-oasis).
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: I think I'd prefer this in a bulleted list

**Comment by edquist** [comment]
@brianhlin, I think I've addressed the comments you've made in this PR.

In Mat's #476, you also mention another step at the end:
>Let the OSG know that the site is ready, ramp up job sub numbers (probably doesn't need its own subsection)

I have not included this step, and could use some help/discussion about what all we want to say there, if we want to add this section too.

Also note that since I've based the current PR off of Mat's PR head commit, merging this PR will also automatically merge his.

File: docs/compute-element/hosted-ce.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+Providing the OSG Worker Node client (HTCondor batch systems only)
+------------------------------------------------------------------
+
+All OSG sites need to provide the worker node client.
+This is normally handled by OSG staff for a Hosted CE, but requires shared home directories across the cluster.
+However, for sites with an HTCondor batch system, often there is no shared filesystem set up.
+If it is easier for your site to install and maintain the worker node client than to set up a shared file system, you have the following options:
+
+-   Install the [osg-wn-client rpm](/worker-node/install-wn)
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
-   Install the [Worker Node Client from RPM](/worker-node/install-wn)
```

File: docs/compute-element/hosted-ce.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+-   Install the [osg-wn tarball](/worker-node/install-wn-tarball)
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
-   Install the [Worker Node Client from tarball](/worker-node/install-wn-tarball)
```

File: docs/compute-element/hosted-ce.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+-   Install the worker node client from [OASIS](/worker-node/install-wn-oasis)
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
-   Install the Worker Node Client from [OASIS](/worker-node/install-wn-oasis)
```

File: docs/compute-element/hosted-ce.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
+in order to support application software repositories.
+This will allow a broader set of jobs to run on your cluster.
+
+
+Providing the OSG Worker Node client (HTCondor batch systems only)
+------------------------------------------------------------------
+
+All OSG sites need to provide the worker node client.
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
All OSG sites need to provide the OSG Worker Node Client on each worker node in the cluster.
```

File: docs/compute-element/hosted-ce.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+This is normally handled by OSG staff for a Hosted CE, but requires shared home directories across the cluster.
+However, for sites with an HTCondor batch system, often there is no shared filesystem set up.
+If it is easier for your site to install and maintain the worker node client than to set up a shared file system, you have the following options:
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
If it is easier for your site to install and maintain the Worker Node Client on each worker node than to set up a shared file system, you have the following options:
```

File: docs/compute-element/hosted-ce.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+-   OSG staff are readily contactable in case of an emergency,
+    through email at <mailto:help@opensciencegrid.org>
+
+
+Applying for an OSG Hosted CE
+-----------------------------
+
+Before making any system changes, you should do the following steps:
+
+-   Fill out the [cluster integration questionnaire](http://goo.gl/forms/8OukxsyG6KBSGHuR2)
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Updated link (unfortunately the short-form seems to expire after some time)

```suggestion
-   Fill out the [cluster integration questionnaire](https://docs.google.com/forms/d/e/1FAIpQLSf_pJ3Xlf2N9v5I1GK9oA9HVIxNeRikEq_T1GY0ijaUMy9cIA/viewform)
```

File: docs/compute-element/hosted-ce.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+    running on a [supported operating system](/release/supported_platforms)
+
+-   Outbound network connectivity from the compute nodes (they can be behind NAT)
+
+-   A Unix account on your cluster's submit server, accessible via an SSH key
+    The OSG CE will use this account to automatically submit jobs,
+    so it must also have permissions to submit jobs to the batch system
+
+-   If your batch system is not HTCondor,
+    there must be a shared file system between the submit server and the compute nodes
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Let's add a `See this section for details` with a link to the section you added below

File: docs/compute-element/hosted-ce.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+An OSG Compute Element (CE) is the entry point for jobs coming from the OSG;
+it handles authorization and delegation of jobs to your existing campus HPC/HTC cluster.
+
+Many sites set up their compute element locally.
+As an alternative, OSG offers a hosted CE option
+wherein the OSG team will host and operate the HTCondor compute element,
+and configure it for the science communities that you choose to support.
+
+This document explains the requirements and the procuedure for obtaining a hosted CE.
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Could you run a spell-checker?

```suggestion
This document explains the requirements and the procedure for obtaining a hosted CE.
```

File: docs/compute-element/hosted-ce.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+------------
+
+An OSG Compute Element (CE) is the entry point for jobs coming from the OSG;
+it handles authorization and delegation of jobs to your existing campus HPC/HTC cluster.
+
+Many sites set up their compute element locally.
+As an alternative, OSG offers a hosted CE option
+wherein the OSG team will host and operate the HTCondor compute element,
+and configure it for the science communities that you choose to support.
+
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Let's add a sentence here saying that the Hosted-CE can support thousands of concurrent job submissions

File: docs/compute-element/hosted-ce.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+This document explains the requirements and the procuedure for obtaining a hosted CE.
+If you wish to have a local compute element instead,
+see page for [installing HTCondor-CE](/compute-element/install-htcondor-ce).
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Let's change this to something like:

> If you wish to run your own local compute element or expect to support more than 10,000 concurrently running OSG jobs, see this page for...

File: docs/compute-element/hosted-ce.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+
+-   Configure the OSG CE with your system details
+
+-   Validate operation with a set of test jobs
+
+-   Configure central OSG services to schedule jobs
+
+Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
+in order to support application software repositories.
+This will allow a broader set of jobs to run on your cluster.
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Link to the section below about OASIS somewhere in this paragraph

File: docs/compute-element/hosted-ce.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+    to set up a consultation call with the OSG team,
+    in order to discuss how you would like to contribute to the OSG;
+    for example, the number of OSG jobs that should run, what resource limits you have,
+    or which science communities you support
+
+After the consultation, do the following:
+
+-   Create a Unix account on the submit server for use by the OSG CE
+
+-   Install public SSH keys for the account
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
-   Install OSG-provided public SSH keys for the account
```

File: docs/compute-element/hosted-ce.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+
+-   Email [help@opensciencegrid.org](mailto:help@opensciencegrid.org)
+    to set up a consultation call with the OSG team,
+    in order to discuss how you would like to contribute to the OSG;
+    for example, the number of OSG jobs that should run, what resource limits you have,
+    or which science communities you support
+
+After the consultation, do the following:
+
+-   Create a Unix account on the submit server for use by the OSG CE
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]:
```suggestion
-   Create a Unix account for OSG jobs on the submit server and across the cluster as necessary
```

File: docs/compute-element/hosted-ce.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+
+-   Install public SSH keys for the account
+
+Once this is done, the OSG team will:
+
+-   Configure the OSG CE with your system details
```
Comment (brianhlin) [inline docs/compute-element/hosted-ce.md]: Let's replace OSG CE with "Hosted CE" everywhere in the doc and make sure Hosted CE is capitalized consistently

**Comment by brianhlin** [comment]
@edquist there are just a few minimal changes left, let's get 'er done

File: docs/compute-element/hosted-ce.md (hunk 19 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: Updated

File: docs/compute-element/hosted-ce.md (hunk 2 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: (updated)

File: docs/compute-element/hosted-ce.md (hunk 12 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: updated

Comment (edquist) [reply docs/compute-element/hosted-ce.md]: updated

File: docs/compute-element/hosted-ce.md (hunk 14 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: updated

File: docs/compute-element/hosted-ce.md (hunk 15 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: updated

File: docs/compute-element/hosted-ce.md (hunk 16 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: updated

File: docs/compute-element/hosted-ce.md (hunk 15 above)
Comment (edquist) [reply docs/compute-element/hosted-ce.md]: updated

**Comment by brianhlin** [comment]
LGTM!

**Pull request merged**